realm/realm-library/src/objectServer/java/io/realm/internal/SyncObjectServerFacade.java

```diff
@@ -206,4 +217,40 @@ public boolean isPartialRealm(RealmConfiguration configuration) {
     public void addSupportForObjectLevelPermissions(RealmConfiguration.Builder builder) {
         builder.addModule(new ObjectPermissionsModule());
     }
+
+    @Override
+    public void downloadInitialSubscriptions(Realm realm) {
```

Comment:
Should we include an optional timeout parameter to cancel the wait, in case all subscriptions could not be retrieved for some reason (error/latency etc.)

Reply:
We probably should, but that should be on the `SyncConfiguration.waitForInitialRemoteData`. I think that is outside the scope of this PR, but I'll create an issue tracking it.